### PR TITLE
Add tests of duplicate object merging behavior

### PR DESCRIPTION
**Pull request by ted-faros**
## Description

Additional unit tests

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: destinations/airbyte-faros-destination/test/__snapshots__/graphql-client.test.ts.snap [hunk 1]
```diff
@@ -65,6 +65,12 @@ Array [
     },
     "uid": "vitalyg",
   },
+  Object {
+    "email": "some-email",
+    "htmlUrl": "https://github.com/jeniii",
+    "name": "some-name",
+    "uid": "jeniii",
+  },
```
Comment (ted-faros) [inline destinations/airbyte-faros-destination/test/__snapshots__/graphql-client.test.ts.snap:73]: shows expected merging behavior in low-level test

File: destinations/airbyte-faros-destination/test/__snapshots__/graphql-client.test.ts.snap [hunk 2]
```diff
@@ -104,6 +110,8 @@ exports[`graphql-client write batch upsert update as upsert 4`] = `"mutation { i
...
 
+exports[`graphql-client write batch upsert upsert same object 1`] = `"mutation { insert_vcs_User (objects: [{uid: \\"jeniii\\", name: \\"some-name\\", email: \\"some-email\\", htmlUrl: \\"https://github.com/jeniii\\", origin: \\"mytestsource\\"}], on_conflict: {constraint: vcs_User_pkey, update_columns: [email, htmlUrl, name, origin, refreshedAt]}) { returning { id refreshedAt source uid } } }"`;
+
```
Comment (ted-faros) [inline destinations/airbyte-faros-destination/test/__snapshots__/graphql-client.test.ts.snap:114]: Shows expected merging behavior in end-to-end test.  Here we see the generated upsert for 2 records from test has non-null values for name and email

**Review by tovbinm — APPROVED** [review]
